src/main/java/jbash/parser/JBashParser.java

```diff
@@ -1,18 +1,44 @@
 package jbash.parser;
 
+import jbash.filesystem.FileSystemAPI;
+
 import java.util.*;
 
 public final class JBashParser {
     private final ArrayList<Token> tokens = new ArrayList<>();
+    private String redirectTo = "";
     JBashParser(ArrayList<Token> tokens) {
         this.tokens.addAll(tokens);
     }
 
+    /**
+     * FIXME: Hacky implementation that will be replaced in its entirety later.
+     *        This is only here so certain filesystem operations can be done, while
+     *        we wait for the parser to be better-implemented.
+     */
+    private void processRedirects() {
+        for (int i = 0; i < tokens.size(); i++) {
+            Token t = tokens.get(i);
+            if (t.type() == TokenType.Great) {
+                if (i + 1 >= tokens.size()) throw new RuntimeException("Expected token after redirect");
+
+                Token next = tokens.get(i + 1);
+                if (next.type() == TokenType.Whitespace) {
```

Comment:
If we assign next to the next token after a whitespace token, the whitespace isn't properly removed.

Reply:
This is a good point actually, and I didn't catch this. Example of for loop constructs being pretty error-prone. This entire method is going in the can one way or another soon enough.